int32 pin
bool state
---
bool active 